packages/fxa-settings/src/pages/ResetPassword/ResetPasswordWithRecoveryKeyVerified/container.tsx

```diff
@@ -45,15 +45,24 @@ const ResetPasswordWithRecoveryKeyVerifiedContainer = ({
 
   const account = useAccount();
   const { uid, sessionToken } = currentAccount() || {};
-  const { keyFetchToken, unwrapBKey } =
-    sensitiveDataClient.getDataType(SensitiveData.Key.AccountReset) || {};
-  const newRecoveryKeyData = sensitiveDataClient.getData('newRecoveryKeyData');
-  const newRecoveryKey = formatRecoveryKey(newRecoveryKeyData.buffer);
-
   const updateRecoveryKeyHint = useCallback(
     async (hint: string) => account.updateRecoveryKeyHint(hint),
     [account]
   );
+  const { keyFetchToken, unwrapBKey } =
+    sensitiveDataClient.getDataType(SensitiveData.Key.AccountReset) || {};
+  const recoveryKey = sensitiveDataClient.getDataType(
+    SensitiveData.Key.NewRecoveryKey
+  )?.recoveryKey;
+
+  if (!recoveryKey) {
```

Comment:
I'm wondering if we should lift the check for `sessionToken` and `uid` out of the `navigateNext` function and above the check for `recoveryKey` - if that data is not in local storage then we might as well go directly to signin